src/Symfony/Component/Form/EnumFormTypeGuesser.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form;
+
+use Symfony\Component\Form\Extension\Core\Type\EnumType as FormEnumType;
+use Symfony\Component\Form\Guess\Guess;
+use Symfony\Component\Form\Guess\TypeGuess;
+use Symfony\Component\Form\Guess\ValueGuess;
+
+final class EnumFormTypeGuesser implements FormTypeGuesserInterface
+{
+    /**
+     * @var array<string, array<string, \ReflectionNamedType|false>>
+     */
+    private array $cache = [];
+
+    public function guessType(string $class, string $property): ?TypeGuess
+    {
+        if (!($propertyType = $this->getPropertyType($class, $property))) {
+            return null;
+        }
+
+        return new TypeGuess(FormEnumType::class, ['class' => $propertyType->getName()], Guess::HIGH_CONFIDENCE);
+    }
+
+    public function guessRequired(string $class, string $property): ?ValueGuess
+    {
+        if (!($propertyType = $this->getPropertyType($class, $property))) {
+            return null;
+        }
+
+        return new ValueGuess(!$propertyType->allowsNull(), Guess::HIGH_CONFIDENCE);
+    }
+
+    public function guessMaxLength(string $class, string $property): ?ValueGuess
+    {
+        return null;
+    }
+
+    public function guessPattern(string $class, string $property): ?ValueGuess
+    {
+        return null;
+    }
+
+    private function getPropertyType(string $class, string $property): \ReflectionNamedType|false
```

Comment:
```suggestion
    private function getPropertyType(string $class, string $property): ?\ReflectionNamedType
```

What about this instead?

Reply:
See https://github.com/symfony/symfony/pull/61297#discussion_r2253809975